types/highland/index.d.ts

```diff
@@ -1204,6 +1372,38 @@ declare namespace Highland {
 		zip(ys: R[]): Stream<R>;
 		zip(ys: Stream<R>): Stream<R>;
 
+		/**
+		 * Takes a stream and a *finite* stream of `N` streams
+		 * and returns a stream of the corresponding `(N+1)`-tuples.
+		 *
+		 * *Note:* This transform will be renamed `zipEach` in the next major version
+		 * release.
+		 *
+		 * @id zipAll
+		 * @section Higher-order Streams
+		 * @name Stream.zipAll(ys)
+		 * @param {Array | Stream} ys - the array of streams to combine values with
+		 * @api public
+		 */
+		zipAll(ys: R[][]): Stream<R[]>;
+		zipAll(ys: Stream<R[]>): Stream<R[]>;
+
+		/**
+		 * Takes a *finite* stream of streams and returns a stream where the first
+		 * element from each separate stream is combined into a single data event,
+		 * followed by the second elements of each stream and so on until the shortest
+		 * input stream is exhausted.
+		 *
+		 * *Note:* This transform will be renamed `zipAll` in the next major version
+		 * release.
+		 *
+		 * @id zipAll0
+		 * @section Higher-order Streams
+		 * @name Stream.zipAll0()
+		 * @api public
+		 */
+		zipAll0(this: Stream<Stream<R>>): Stream<R[]>;
```

Comment:
I tried to add a test case for zipAll0 but typescript complained - do you know why that might be? The method definition looks right on first glance to me

```
fooStreamStream.zipAll0()
```

```
The 'this' context of type 'Stream<Stream<Foo>>' is not assignable to method's 'this' of type 'Stream<Stream<Stream<Foo>>>'.
  Types of property 'write' are incompatible.
    Type '(x: Stream<Foo>) => boolean' is not assignable to type '(x: Stream<Stream<Foo>>) => boolean'.
      Types of parameters 'x' and 'x' are incompatible.
        Type 'Stream<Stream<Foo>>' is not assignable to type 'Stream<Foo>'.
          Types of property 'write' are incompatible.
            Type '(x: Stream<Foo>) => boolean' is not assignable to type '(x: Foo) => boolean'.
              Types of parameters 'x' and 'x' are incompatible.
                Type 'Foo' is not assignable to type 'Stream<Foo>'.ts(2684)
```

Reply:
Oh yeah, that's an interesting mistake on my part. I think that as written, `this` must be both `Stream<R>` (from the class definition) and `Stream<Stream<R>>` (from the method definition). I tried switching it to
```
zipAll0<T>(this: Stream<Stream<T>>): Stream<T[]>;
```
and the test started to pass.

Sorry for submitting broken code; my typescript codebase doesn't use much highland now, so I don't have real code to test most of this against. More reason I should have been adding test cases for all the new methods.

Reply:
Interesting, I haven't seen that kind of error before, nice work debugging it!